src/radical/pilot/agent/launch_method/mpiexec.py

```diff
@@ -97,24 +120,38 @@ def construct_command(self, cu, launch_script_hop):
         arg_max = 4096
 
         # This is the command w/o the host string
-        command_stub = "%s %%s %s %s" % (self.launch_command,
-                                         env_string, task_command)
+        omplace = ''
+        if self._omplace:
+            omplace = 'omplace'
+        command_stub = "%s %%s %s %s %s" % (self.launch_command,
+                                            env_string, omplace, task_command)
 
         # cluster hosts by number of slots
         host_string = ''
-        for node,nslots in list(host_slots.items()):
-            host_string += '-host %s -n %s ' % (','.join([node] * nslots), nslots)
+        if not self._mpt:
+            for node,nslots in list(host_slots.items()):
+                host_string += '-host '
+                host_string += '%s -n %s ' % (','.join([node] * nslots), nslots)
+        else:
+            slots = 0
+            hosts = list()
+            for node,nslots in list(host_slots.items()):
+                slots +=          nslots
+                hosts += [node] * nslots
+            host_string += ','.join(hosts)
+            host_string += ' -n 1'
```

Comment:
1) Is variable `slots` needed here? (also variable `slots` already exists, even it is not used later, and this operation reassigns it);
2) `host_string` here doesn't need `-host`, right? (didn't find about it in the documentation..)

Reply:
1: yeah, you are right - removed
2: correct, the MPT version doesn't like `-host` ...